### PR TITLE
Adding Icon+Panel to Facebook Containers

### PR DESCRIPTION
**Pull request by ShivangiKakkar**
### Fixes

### CODE REVIEW

**Comment by ShivangiKakkar** [comment]
Messed up `background.js` file I just wrote the upper logic. Working on it.

File: src/background.js [hunk 1]
```diff
@@ -372,6 +384,14 @@ async function containFacebook (options) {
   return {cancel: true};
 }
 
+function showPanel() {
+  if(isFacebookURL){
```
Comment (jonathanKingston) [inline src/background.js]: Should this be accepting a URL? I'm not sure why these if branches are the same either.

File: src/background.js [hunk 2]
```diff
@@ -372,6 +382,14 @@ async function containFacebook (options) {
...
+    browser.browserAction.getPopup({});
+  }else {
```
Comment (jonathanKingston) [inline src/background.js]: Nit: indenting

File: src/panel1.html [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<!DOCTYPE html>
+<html id="1">
+  <head>
+    <meta http-equiv="content-type" content="text/html; charset=utf-8">
+    <title>Facebook Container</title>
+    <link rel="stylesheet" href="panel.css">
+    <script src="background.js"></script>
```
Comment (stoically) [inline src/panel1.html]: The background script should not be included in the popup (same for `panel2.html`) - that's also why some tests are failing (code is executed twice).

File: src/background.js (hunk 1 above)
Comment (stoically) [reply src/background.js]: Also `getPopup` returns the FQDN to the popup panel file and the return value isn't used, so currently the `showPanel` function does nothing.

File: src/background.js [hunk 4]
```diff
@@ -372,6 +382,14 @@ async function containFacebook (options) {
   return {cancel: true};
 }
 
+function showPanel() {
+  if(isFacebookURL){
+    browser.browserAction.getPopup({});
```
Comment (groovecoder) [inline src/background.js]: This is probably trying to call [`browserAction.openPopup()`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/browserAction/openPopup), but there's a problem: "You can only call this function from inside the handler for a user action."

So, we can't force the browserAction popup to open as soon as the user visits facebook.com.

As an alternative, in the #contain-facebook Slack channel we discussed [adding a badge](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/browserAction/setBadgeTextColor) to the `browserAction` icon to visually encourage the user to click the icon and receive the pop-up.

I'm starting a branch based on this one which starts to add this.

**Comment by groovecoder** [comment]
Superseded by https://github.com/mozilla/contain-facebook/pull/214

**Pull request closed without merging**